examples/pytorch/gat/test_gat.py

```diff
@@ -2,99 +2,146 @@
 # Licensed under the MIT License.
 
 import os
+import platform
 import sys
 import pytest
 import tempfile
-import argparse
 import numpy as np
 import torch
+import argparse
 
-from deepgnn import get_logger
-from deepgnn.pytorch.common.utils import set_seed
 from deepgnn.pytorch.common.dataset import TorchDeepGNNDataset
 from deepgnn.graph_engine import (
     GraphType,
     BackendType,
-    BackendOptions,
     FileNodeSampler,
+    BackendOptions,
     create_backend,
 )
 from deepgnn.graph_engine.snark.converter.options import DataConverterType
 from deepgnn.graph_engine.data.citation import Cora
-from model import GAT, GATQueryParameter  # type: ignore
 
+from model_geometric import GAT, GATQueryParameter  # type: ignore
+from main import run_ray  # type: ignore
+from deepgnn import get_logger
+
+
+def setup_module(module):
+    import deepgnn.graph_engine.snark._lib as lib
```

Comment:
why do we need this extra setting for Ray?

Reply:
We were not using GE in this test before so we did not need it, instead were using mock dataloader.

Reply:
why do we need to start using GE and drop the mock dataloader?

Reply:
mock dataloader really needs to be fully removed and replaced with GE usage so the unit tests are simpler and the models are robust to real datasets. I removed as much as possible in this PR.

Reply:
Swapping backends should be easy, otherwise it means a design flaw.
Please don't make unrelated changes to Ray in this PR.

Reply:
It is not an unrelated change, I made the unit tests use the same code as run.sh which is why I was able to remove MockDataLoader without extra work.

Reply:
Also mockdataloader is not just a different backend, a lot of the functions are implemented improperly so it affects model design and test outcome, it should be removed completely in another PR.